internal/literals/numbers.go

```diff
@@ -0,0 +1,326 @@
+			X:  convertExpr("uint32", indexExpr("data", intLiteral("3"))),
+			Op: token.SHL,
+			Y:  intLiteral("24"),
+		},
+	}
+
+	block.List = append(block.List, boundsCheckData("3"), returnStmt(convertExpr))
+
+	return lambdaCall(ident("uint32"), block)
+}
+
+func obfuscateUint64(data uint64) *ast.CallExpr {
+	obfuscator := randObfuscator()
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, data)
+	block := obfuscator.obfuscate(b)
+	convertExpr := &ast.BinaryExpr{
+		X: &ast.BinaryExpr{
+			X: &ast.BinaryExpr{
+				X: &ast.BinaryExpr{
+					X: &ast.BinaryExpr{
+						X: &ast.BinaryExpr{
+							X: &ast.BinaryExpr{
+								X:  convertExpr("uint64", indexExpr("data", intLiteral("0"))),
+								Op: token.OR,
+								Y: &ast.BinaryExpr{
+									X:  convertExpr("uint64", indexExpr("data", intLiteral("1"))),
+									Op: token.SHL,
+									Y:  intLiteral("8"),
+								},
+							},
+							Op: token.OR,
+							Y: &ast.BinaryExpr{
+								X:  convertExpr("uint64", indexExpr("data", intLiteral("2"))),
+								Op: token.SHL,
+								Y:  intLiteral("16"),
+							},
+						},
+						Op: token.OR,
+						Y: &ast.BinaryExpr{
+							X:  convertExpr("uint64", indexExpr("data", intLiteral("3"))),
+							Op: token.SHL,
+							Y:  intLiteral("24"),
+						},
+					},
+					Op: token.OR,
+					Y: &ast.BinaryExpr{
+						X:  convertExpr("uint64", indexExpr("data", intLiteral("4"))),
+						Op: token.SHL,
+						Y:  intLiteral("32"),
+					},
+				},
+				Op: token.OR,
+				Y: &ast.BinaryExpr{
+					X:  convertExpr("uint64", indexExpr("data", intLiteral("5"))),
+					Op: token.SHL,
+					Y:  intLiteral("40"),
+				},
+			},
+			Op: token.OR,
+			Y: &ast.BinaryExpr{
+				X:  convertExpr("uint64", indexExpr("data", intLiteral("6"))),
+				Op: token.SHL,
+				Y:  intLiteral("48"),
+			},
+		},
+		Op: token.OR,
+		Y: &ast.BinaryExpr{
+			X:  convertExpr("uint64", indexExpr("data", intLiteral("7"))),
+			Op: token.SHL,
+			Y:  intLiteral("56"),
+		},
+	}
+
+	block.List = append(block.List, boundsCheckData("7"), returnStmt(convertExpr))
+
+	return lambdaCall(ident("uint64"), block)
+}
+
+func obfuscateUint(data uint) *ast.CallExpr {
+	return convertExpr("uint", obfuscateUint64(uint64(data)))
+}
+
+func obfuscateUintptr(data uintptr) *ast.CallExpr {
+	return convertExpr("uintptr", obfuscateUint64(uint64(data)))
+}
+
+func obfuscateInt8(data int8) *ast.CallExpr {
+	return convertExpr("int8", obfuscateUint8(uint8(data)))
+}
+
+func obfuscateInt16(data int16) *ast.CallExpr {
+	return convertExpr("int16", obfuscateUint16(uint16(data)))
+}
+
+func obfuscateInt32(data int32) *ast.CallExpr {
+	return convertExpr("int32", obfuscateUint32(uint32(data)))
+}
+
+func obfuscateInt64(data int64) *ast.CallExpr {
+	return convertExpr("int64", obfuscateUint64(uint64(data)))
+}
+
+func obfuscateInt(data int) *ast.CallExpr {
+	return convertExpr("int", obfuscateUint64(uint64(data)))
+}
+
+func uintToFloat(uintExpr *ast.CallExpr, bits string) *ast.CallExpr {
```

Comment:
I'm pretty sure this could be just calling https://golang.org/pkg/math/#Float64frombits